src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class ConvertSwitchStatementToExpressionCodeFixProvider
+    {
+        private sealed class Rewriter : CSharpSyntaxVisitor<ExpressionSyntax>
+        {
+            private readonly ArrayBuilder<ExpressionSyntax> _assignmentTargets;
+
+            private Rewriter()
+            {
+                _assignmentTargets = ArrayBuilder<ExpressionSyntax>.GetInstance();
+            }
+
+            private void Free()
+            {
+                _assignmentTargets.Free();
+            }
+
+            public static StatementSyntax Rewrite(SwitchStatementSyntax node, SemanticModel semanticModel, SyntaxEditor editor)
+            {
+                var rewriter = new Rewriter();
+                var switchExpression = rewriter.VisitSwitchStatement(node);
+                var generateDeclaration = rewriter.TryRemoveVariableDeclarators(node, semanticModel, editor);
+                var finalStatement = rewriter.GetFinalStatement(switchExpression, generateDeclaration);
+                rewriter.Free();
+                return finalStatement;
+            }
+
+            private bool TryRemoveVariableDeclarators(SwitchStatementSyntax node, SemanticModel semanticModel, SyntaxEditor editor)
```

Comment:
instead of node, can you call this switchStatement.  it will make later code in this method easier to understand.  thanks!